include/openthread/platform/crypto.h

```diff
@@ -480,15 +480,88 @@ void otPlatCryptoRandomDeinit(void);
 /**
  * Fills a given buffer with cryptographically secure random bytes.
  *
- * @param[out] aBuffer  A pointer to a buffer to fill with the random bytes.
- * @param[in]  aSize    Size of buffer (number of bytes to fill).
+ * @param[out] aBuffer            A pointer to a buffer to fill with the random bytes.
+ * @param[in]  aSize              Size of buffer (number of bytes to fill).
  *
- * @retval OT_ERROR_NONE     Successfully filled buffer with random values.
- * @retval OT_ERROR_FAILED   Operation failed.
+ * @retval OT_ERROR_NONE          Successfully filled buffer with random values.
+ * @retval OT_ERROR_FAILED        Operation failed.
  *
  */
 otError otPlatCryptoRandomGet(uint8_t *aBuffer, uint16_t aSize);
 
+/**
+ * Generate and populate the output buffer with a new key-pair.
+ *
+ * @param[out] aBuffer            A pointer to a buffer to store the generated key-pair.
+ * @param[in]  aInLen             Size of the input buffer.
+ * @param[out] aOutLen            A pointer to a variable to store the size of the generated key-pair.
+ *
+ * @retval OT_ERROR_NONE          A new key-pair was generated successfully.
+ * @retval OT_ERROR_NO_BUFS       Failed to allocate buffer for key generation.
+ * @retval OT_ERROR_NOT_CAPABLE   Feature not supported.
+ * @retval OT_ERROR_FAILED        Failed to generate key-pair.
+ *
+ */
+otError otPlatCryptoEcdsaGenerate(uint8_t *aBuffer, uint8_t aInLen, uint8_t *aOutLen);
+
+/**
+ * Parse the input key-pair in DER format and generate a key-pair context.
+ *
+ * @param[out] aContext           A pointer to a context to store the key-pair.
+ * @param[in]  aBuffer            A pointer to a buffer where the key-pair is stored in DER format.
+ * @param[in]  aSize              The size of the input buffer.
+ *
+ * @retval OT_ERROR_NONE          The key was retrieved successfully, and @p aContext is updated.
+ * @retval OT_ERROR_PARSE         The key-pair DER format could not be parsed (invalid format).
+ *
+ */
+otError otPlatCryptoEcdsaParse(void *aContext, const uint8_t *aBuffer, uint8_t aSize);
```

Comment:
I don't think this make sense as platform API. It is mentioning to parse a key-pair in `aBuffer` (in DER format) into a `aContext` format which is not defined here. 

We should consider that a platform developer should be able to read the documentations here and know what they need to implement.

This seem to mirror `KeyPair::Parse()` method which is `private` and used by other methods in `KeyPair`. We can do the same here (meaning no need to expose this as an API).

Reply:
I guess we can share the common code (e.g. for parsing) in the implementation of `otPlatCryptoEcdsa` functions (under different flavors: mbedtls or tinycrypt).

I think in new model with `otPlatCrypto` APIs we won't need or have the `KeyPair::Parse()` anymore? The `KeyPair` methods can just call the related `otPlatCryptoEcdsa{}` function.

Reply:
Do you mean to get rid of `KeyPair::Parse()` completely? I see that it's used only in `KeyPair::GetPublicKey` and `KeyPair::Sign`. It does duplicate the code a bit but it's not much.

Reply:
Yes. When we go for `otPLatCryptoEcdsa` model the `KeyPair::GetPublicKey()` will just calls `otPlatCryptoEcdsaGePublicKey()` (same with all other methods `ecdsa.hpp`).  We are basically moving all implementation to platform layer.

Reply:
Got it. Will start the work soon. Thank you for the review.